near-sdk/src/store/lookup_set/mod.rs

```diff
@@ -92,7 +167,38 @@ where
     where
         T: Clone,
     {
-        self.map.insert(value, ()).is_none()
+        let entry = self.get_mut_inner(&value);
+        match entry {
+            EntryState::Inserted | EntryState::Present => false,
+            EntryState::Deleted => {
+                let _ = std::mem::replace(entry, EntryState::Present);
+                true
+            }
+            EntryState::Absent => {
+                let _ = std::mem::replace(entry, EntryState::Inserted);
+                true
+            }
+        }
+    }
+
+    /// Puts the given value into the set.
+    ///
+    /// This function will not return whether the passed value was already in the set.
+    /// Use [`LookupSet::insert`] if you need that.
+    pub fn put(&mut self, value: T) {
+        let entry = self.cache.get_mut(value);
+        match entry.get_mut() {
+            Some(EntryState::Inserted) | Some(EntryState::Present) => {}
+            Some(EntryState::Deleted) => {
+                let _ = entry.set(EntryState::Present);
+            }
+            Some(EntryState::Absent) | None => {
+                // It is safe to mark an entry as `Inserted` even if it is already present on trie;
+                // it just means we will invoke `env::storage_write` one more time than strictly
+                // necessary.
+                let _ = entry.set(EntryState::Inserted);
```

Comment:
these sets will fail to change anything in every case except `None`

Here is one test you can add which will verify this:
```rust
    #[test]
    fn check_put_on_deleted() {
        let mut set = LookupSet::new(b"m");
        assert!(set.insert(8u8));
        assert!(set.remove(&8));
        set.put(8);
        assert!(set.contains(&8));
    }
```

This is why you should avoid using `let _ =` unless you are sure it cannot error. When building it out it's just safer to unwrap these to make sure you're not going to have one of these types of bugs that are hard to find

Reply:
Oh, I didn't realize that `OnceCell::set` returns `Result`... I thought it returns the previously set value and that the compiler was just complaining about me ignoring the output of the function (that I didn't need).

Reply:
there wouldn't be a need to use `let _ = ` if it just returned the previous type as this should only ever be used to ignore `#[must_use]` types like `Result` when you're sure of it. 

If you could overwrite values of OnceCell it would be unsafe because when changed, the references held of the value would be invalidated, if you're curious of slightly more specifics of the API.

Reply:
Makes sense, thanks for the explanation.